File: storage/src/main/java/com/snatik/storage/helpers/SizeUnit.java
package com.snatik.storage.helpers;

import java.text.DecimalFormat;

public enum SizeUnit {
    B(1),
    KB(SizeUnit.BYTES),
    MB(SizeUnit.BYTES * SizeUnit.BYTES),
    GB(SizeUnit.BYTES * SizeUnit.BYTES * SizeUnit.BYTES),
    TB(SizeUnit.BYTES * SizeUnit.BYTES * SizeUnit.BYTES * SizeUnit.BYTES);

    private long inBytes;
    private static final int BYTES = 1024;

    private SizeUnit(long bytes) {
        this.inBytes = bytes;
    }

    public long inBytes() {
        return inBytes;
    }

    public static String readableSizeUnit(long bytes) {
        DecimalFormat df = new DecimalFormat("0.00");
        if (bytes < KB.inBytes()) {
            return df.format(bytes / (float) B.inBytes()) + " B";
        } else if (bytes < MB.inBytes()) {
            return df.format(bytes / (float) KB.inBytes()) + " KB";
        } else if (bytes < GB.inBytes()) {
            return df.format(bytes / (float) MB.inBytes()) + " MB";
        } else {
            return df.format(bytes / (float) GB.inBytes()) + " GB";
        }
    }
}

File: storage/src/main/java/com/snatik/storage/helpers/SizeUnit.kt
package com.snatik.storage.helpers

import java.text.DecimalFormat

const val BYTES = 1024L
enum class SizeUnit(private val inBytes: Long) {
    B(1),
    KB(BYTES),
    MB(BYTES * BYTES),
    GB(BYTES * BYTES * BYTES),
    TB(BYTES * BYTES * BYTES * BYTES);

    fun inBytes(): Long {
        return inBytes
    }

    companion object {
        fun readableSizeUnit(bytes: Long): String {
            val df = DecimalFormat("0.00")
            return when {
                bytes < KB.inBytes() -> df.format((bytes / B.inBytes().toFloat()).toDouble()) + " B"
                bytes < MB.inBytes() -> df.format((bytes / KB.inBytes().toFloat()).toDouble()) + " KB"
                bytes < GB.inBytes() -> df.format((bytes / MB.inBytes().toFloat()).toDouble()) + " MB"
                else -> df.format((bytes / GB.inBytes().toFloat()).toDouble()) + " GB"
            }
        }
    }
}
